### PR TITLE
Replace switch-specific code with env sensor api; Add device light sensor option

### PR DESCRIPTION
**Pull request by p-sam**
* Removes gyro and accel handling code that was specific to switch, it now uses the sensors via libretro interfaces.
* Adds a "sensor" value to the light sensor core option, which allows to use the device sensor instead of fixed values
* Fixes setting the light sensor core option would not update the level ingame

### CODE REVIEW

File: src/platform/libretro/libretro_core_options.h [hunk 1]
```diff
@@ -54,18 +54,18 @@ struct retro_core_option_definition option_defs_us[] = {
...
-         { "3",  NULL },
-         { "4",  NULL },
-         { "5",  NULL },
-         { "6",  NULL },
-         { "7",  NULL },
-         { "8",  NULL },
-         { "9",  NULL },
-         { "10", NULL },
-         { NULL, NULL },
+         { "sensor",  "Use device sensor if available" },
```
Comment (endrift) [inline src/platform/libretro/libretro_core_options.h]: Shouldn't the 0 option still be here?

Comment (p-sam) [reply src/platform/libretro/libretro_core_options.h]: Oh yeah it's missing, fixing

**Comment by endrift** [comment]
I can't review this it seems but I think it would be valuable to merge this.

**Comment by p-sam** [comment]
@twinaphex Hello, would you be available to review this?

**Comment by p-sam** [comment]
rebased with master

**Comment by twinaphex** [comment]
Took a very long time but this is finally merged now. My bad. A PR by you guys will never take this long again from now on.

**Pull request merged**